src/Lecture2.hs

```diff
@@ -208,8 +231,9 @@ The algorithm of merge sort is the following:
 [1,2,3]
 -}
 mergeSort :: [Int] -> [Int]
-mergeSort = error "TODO"
-
+mergeSort []       = []
+mergeSort [x]      = [x]
+mergeSort (x : xs) = merge [x] (mergeSort xs)
```

Comment:
That's a clever and correct solution 😅 
Unfortunately, it's not optimal asymptotically. Your solution has complexity `O(n ^ 2)` while optimal merge sort has `O(n log n)`.